config/initializers/0_ldap_authenticatable.rb

```diff
@@ -0,0 +1,109 @@
+require 'net/ldap'
+require 'devise/strategies/authenticatable'
+
+module Devise
+  module Strategies
+    class LdapAuthenticatable < Authenticatable
+      def valid?
+        Rails.configuration.x.use_ldap && params[:user].present?
+      end
+
+      def authenticate!
+        config = Rails.configuration
```

Comment:
I believe it is probably okay to simply read the ENV variables in this very class, instead of doing so in an initializer. If nothing but this class will use them.